src/main/java/com/google/songgame/servlets/GameServlet.java

```diff
@@ -48,19 +54,128 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
-    SpotifyApi spotifyApi = new SpotifyApi.Builder().setClientId(clientId).setClientSecret(clientSecret).build();
-    ClientCredentialsRequest clientCredentialsRequest = spotifyApi.clientCredentials().build();
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String url = getParameter(request, "playlist-link", "");
+    setVideoID(url);
+    response.sendRedirect("/game.html");
+  }
+  
+  /**
+   * @return the request parameter, or the default value if the parameter was not specified by the
+   *     client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
 
+  /**
+   * Use url to connect to YouTube API and set video ID.
+   *
+   * @param url
+   */
+  public void setVideoID(String url) {
+    String playlistID = getIDFromURL(url);
+    PlaylistItemListResponse playlistItem = new PlaylistItemListResponse();
+    // Retrieve Playlist item from Youtube API
     try {
-      ClientCredentials clientCredentials = clientCredentialsRequest.execute();
-      return clientCredentials.getAccessToken();
-    } catch (IOException | SpotifyWebApiException | ParseException e) {
-      System.out.println("Error: " + e.getMessage());
+      playlistItem = getPlaylistInfo(playlistID);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    // Parse Playlist item Json string to retrieve video IDs
+    String playlistItemJson = new Gson().toJson(playlistItem);
+    ArrayList<String> playlistVideos = parsePlaylistItem(playlistItemJson);
+    PLAYLIST_SIZE = playlistVideos.size();
+    videoID = getRandomVideo(playlistVideos);
+  }
+
+  /**
+   * Parse Youtube URL for playlist ID.
+   *
+   * @param url
+   * @return playlist ID
+   */
+  public String getIDFromURL(String url) {
+    if (url.contains("youtube.com/playlist?list=")) {
+      int start = url.indexOf("list=") + 5;
+      int end = url.length();
+      if (url.contains("&")) {
+        end = url.indexOf("&", start);
+      }
+      return url.substring(start, end);
+    } else {
+      System.err.println(url + " is not a valid YouTube Playlist URL");
+      return "";
+      // TODO @hdee: do something more complicated to handle this error
     }
-    return "";
   }
 
+    /**
+   * Call function to create API service object.
+   *
+   * @throws GeneralSecurityException, IOException, GoogleJsonResponseException
+   */
+  public static PlaylistItemListResponse getPlaylistInfo(String playlistID)
+      throws GeneralSecurityException, IOException, GoogleJsonResponseException {
+    YouTube youtubeService = getService();
+    // Define and execute the API request
+    YouTube.PlaylistItems.List request =
+        youtubeService
+            .playlistItems()
+            .list(Arrays.asList("snippet"))
+            .setMaxResults(25L)
```

Comment:
Might be useful to change 25L to a constant defined either in the function or class.